app/cluster-api-provider-sidero/controllers/metalmachine_controller.go

```diff
@@ -215,6 +215,7 @@ func (r *MetalMachineReconciler) reconcileDelete(ctx context.Context, metalMachi
 		}
 
 		serverResource.Status.InUse = false
```

Comment:
I'm torn on leaving this here. Technically, the server is no longer in use in the sense that it is part of a cluster. On the other hand we run the risk of accidentally reusing the server before we have a chance to reset it. The alternative is that we remove this line, and mark the server as not in use when the agent updates the `NeedsWipe` (after it resets the node). Thoughts?

Reply:
Should we search for !InUser and !NeedsWipe servers then?